kubectl-cloudflow/verify/util_test.go

```diff
@@ -0,0 +1,189 @@
+
+func Test_ParseImageReferenceNoRepo(t *testing.T) {
+	imageRef, _ := ParseImageReference("image:test")
+	assert.Equal(t, "", imageRef.Registry)
+	assert.Equal(t, "", imageRef.Repository)
+	assert.Equal(t, "image", imageRef.Image)
+	assert.Equal(t, "test", imageRef.Tag)
+}
+
+func Test_ParseImageReferenceNoRegistry(t *testing.T) {
+	imageRef, _ := ParseImageReference("grafana/grafana:test")
+	assert.Equal(t, "", imageRef.Registry)
+	assert.Equal(t, "grafana", imageRef.Repository)
+	assert.Equal(t, "grafana", imageRef.Image)
+	assert.Equal(t, "test", imageRef.Tag)
+}
+
+func Test_ParseImageReferenceSha(t *testing.T) {
+	imageRef, _ := ParseImageReference("eu.gcr.io/dummy/lightbend/rays-sensors@sha256:0840ebe4b207b9ca7eb400e84bf937b64c3809fc275b2e90ba881cbecf56c39a")
+	assert.Equal(t, "eu.gcr.io", imageRef.Registry)
+	assert.Equal(t, "dummy/lightbend", imageRef.Repository)
+	assert.Equal(t, "rays-sensors", imageRef.Image)
+	assert.Equal(t, "sha256:0840ebe4b207b9ca7eb400e84bf937b64c3809fc275b2e90ba881cbecf56c39a", imageRef.Tag)
+}
+
+func Test_ParseClassName(t *testing.T) {
+	assert.Equal(t, CheckFullPatternMatch("$0000", ClassNamePattern), true)
+	assert.Equal(t, CheckFullPatternMatch("ab-", ClassNamePattern), false)
+	assert.Equal(t, CheckFullPatternMatch("-ab", ClassNamePattern), false)
+	assert.Equal(t, CheckFullPatternMatch("2ab", ClassNamePattern), false)
+	assert.Equal(t, CheckFullPatternMatch("a/b", ClassNamePattern), false)
+	assert.Equal(t, CheckFullPatternMatch("a+b", ClassNamePattern), false)
+	assert.Equal(t, CheckFullPatternMatch("0000", ClassNamePattern), false)
+	assert.Equal(t, CheckFullPatternMatch("$0000.erere_$", ClassNamePattern), true)
+	assert.Equal(t, CheckFullPatternMatch("__2rererere.$", ClassNamePattern), true)
+	assert.Equal(t, CheckFullPatternMatch("@@@@@@$$$$__", ClassNamePattern), false)
+}
+
+func Test_ParseLabelName(t *testing.T) {
+	assert.Equal(t, IsDnsLabelCompatible("$0000"), false)
+	assert.Equal(t, IsDnsLabelCompatible("0000www"), true)
+	assert.Equal(t, IsDnsLabelCompatible("rfr@3"), false)
+	assert.Equal(t, IsDnsLabelCompatible("__2rererere."), false)
+	assert.Equal(t, IsDnsLabelCompatible("2212dwdwee__"), false)
+}
+
+func Test_ParseConfigurationParameter(t *testing.T) {
+	assert.Equal(t, CheckFullPatternMatch("0000", ConfigParameterKeyPattern), false)
+	assert.Equal(t, CheckFullPatternMatch("$0000", ConfigParameterKeyPattern), false)
+	assert.Equal(t, CheckFullPatternMatch("ER-1qw", ConfigParameterKeyPattern), true)
+	assert.Equal(t, CheckFullPatternMatch("2weee", ConfigParameterKeyPattern), false)
+	assert.Equal(t, CheckFullPatternMatch("configuration.", ConfigParameterKeyPattern), false)
+}
+
+func Test_Sliding(t *testing.T) {
+	var descs= []StreamletDescriptor{randomStreamlet(), randomStreamlet(), randomStreamlet()}
+	var expected= [][]StreamletDescriptor{{descs[0], descs[1]}, {descs[1], descs[2]}}
+	assert.Equal(t, reflect.DeepEqual(expected, sliding(descs)), true)
+}
+
+func Test_Diff(t *testing.T) {
```

Comment:
Should have some more tests for this

Reply:
for instance where the numbers are not sorted, a empty, b empy,  both empty, diff between arrays that do not share any element,  non-consecutive duplicates, ...

Reply:
no need for now. It should be done if we start re-using it.

Reply:
I am refactoring some stuff so I might put it somewhere else.